### PR TITLE
UXD-1079 inline editing status

### PR DESCRIPTION
**Pull request by nahumzs**
### Purpose 🚀
implementing different status for inline-editors

Screen Shot 2021-04-23 at 3 33 39 PM
Screen Shot 2021-04-23 at 3 34 00 PM
Screen Shot 2021-04-23 at 3 36 18 PM

This PR adds the ability to control each cell and set its status when data has been submitted:

Examples

```jsx
onSubmit={(index, options, _, { rowIndex }) => {
  const nextValue = options[index].value;
  
  if (nextValue !== props.row.yourValue) {
    setOptimisticValue(nextValue); // this set the value optimistically it can be ignore

	// three status
	// statusType.IDLE
	// statusType.LOADING
	// statusType.ERROR
	// statusType.SUCCESS (this is transitory, it will revert to IDLE after the animation finished)
	    
    setStatus(statusType.LOADING);

    setData(prevData => {
      const nextData = prevData.slice(0); // your logic
      nextData[rowIndex].subscription = nextValue; // your logic
      setStatus(statusType.SUCCEED); // set the ✅ mark
      return nextData; // your logic
    });
  }
}

```

in case you want to mark the cell with an error:

```jsx
onSubmit={(index, options, _, { rowIndex }) => {
  const nextValue = options[index].value;
  
  if (nextValue !== props.row.yourValue) { 	    
    setStatus(statusType.ERROR, { messageError: "Your message" });
  }
}

```

### Notes ✏️
_details of code change / secondary purposes of this PR_

### Updates 📦
If you have changed a component's source code (not stories, specs, or docs), before merging your branch run `yarn changeset`. This will prompt you to:
- indicate if changes were patch/minor/major for each modified package
- enter a release message

### Storybook 📕
http://storybooks.highbond-s3.com/paprika/UXD-1079-loading-state-inline-editing/?path=/story/forms-inlineeditors--with-in-a-table

### Screenshots 📸
_optional but highly recommended_

### References 🔗
_relevant Jira ticket / GitHub issues_

### CODE REVIEW

File: packages/InlineEditors/src/Editor/Editor.js [hunk 1]
```diff
@@ -1,10 +1,44 @@
...
+      <Popover.Trigger>
+        {(handler, a11yAttributes) => {
+          return Icon ? (
+            <Icon
+              onMouseOver={handler}
+              onMouseOut={handler}
+              onFocus={handler}
+              onBlur={handler}
+              tabIndex={0}
+              aria-label="info"
```
Comment (allison-c) [inline packages/InlineEditors/src/Editor/Editor.js:25]: missing translation

File: packages/InlineEditors/src/Editor/Editor.js [hunk 2]
```diff
@@ -17,9 +51,34 @@ const Editor = React.forwardRef((props, ref) => {
...
+      {status === statusTypes.LOADING ? (
+        <sc.Loading>
+          <Spinner size={Spinner.types.size.SMALL} />
+        </sc.Loading>
+      ) : null}
+      {status === statusTypes.SUCCEED ? (
+        <sc.Success onAnimationEnd={onAnimationEndSuccess}>
+          <Check />
+        </sc.Success>
+      ) : null}
```
Comment (allison-c) [inline packages/InlineEditors/src/Editor/Editor.js:80]: I usually prefer `switch () {case}` for the cases like these, but I guess up to you

**Comment by allison-c** [comment]
I saw a white tooltip instead of the black one, is this expected? 
image

**Comment by nahumzs** [comment]
@allison-c there are some minor tweaks for UI perspective that needs to be happening in an incoming PR, this in specific need to be white and the text colour should be red, but at this point, I'm not focusing too much in those small details, I'm trying to have a stable API that can be controlled, and unblock the team that need this feature. :) 

I will add those UI details in a jira ticket `fyi`

File: packages/InlineEditors/src/Editor/Editor.js [hunk 3]
```diff
@@ -33,15 +92,29 @@ Editor.Edit = () => {
...
   children: PropTypes.node.isRequired,
   isEditing: PropTypes.bool.isRequired,
+  onAnimationEndSuccess: PropTypes.func,
   onClick: PropTypes.func,
+  optimisticValue: PropTypes.node,
+  messageError: PropTypes.node,
 };
 
 const defaultProps = {
+  onAnimationEndSuccess: () => {},
```
Comment (allison-c) [inline packages/InlineEditors/src/Editor/Editor.js:113]: I think this name is a little bit confusing, what about `onSuccessAnimationEnd`? why we need this callback? there's something that has to happen after the green checkmark disappears?

Comment (nahumzs) [reply packages/InlineEditors/src/Editor/Editor.js:113]: yes! we have to revert the internal state into IDLE state again so is possible to edit again the ListBox (or any other element) and not show the Success (check icon) :)

Comment (nahumzs) [reply packages/InlineEditors/src/Editor/Editor.js:113]: I will change the name

**Comment by nahumzs** [comment]
will follow with ui polish and https://github.com/acl-services/paprika/pull/1022#discussion_r620626460 over jira https://aclgrc.atlassian.net/browse/UXD-1100

**Review by allison-c — APPROVED** [review]

**Review by allison-c — CHANGES_REQUESTED** [review]
wait

**Review by allison-c — APPROVED** [review]

**Pull request merged**